packages/botonic-plugin-google-analytics-v4/src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { PluginPostRequest, PluginPreRequest } from '@botonic/core'
+import axios from 'axios'
+
+import { BotSession, GA4Event, GA4Options } from './types'
+
+const defaultGetClientId = (session: BotSession) => session.user.id
+
+export default class BotonicPluginGoogleAnalytics4 {
+  private baseUrl = 'https://www.google-analytics.com/mp/collect'
+  private apiSecret: string
+  private measurementId: string
+  private getClientId: (session: BotSession) => string
+  private userId?: string
+  constructor(options: GA4Options) {
+    this.apiSecret = getEnv('ga4ApiSecret')
+    this.measurementId = getEnv('ga4MeasurementId')
```

Comment:
These `getEnv` methods shouldn't be used and get the values directly from `options` (to be set when defining the plugin in the bot).